lib/cf-socket.c

```diff
@@ -1602,6 +1609,12 @@ CURLcode Curl_cf_tcp_create(struct Curl_cfilter **pcf,
     result = CURLE_OUT_OF_MEMORY;
     goto out;
   }
+
+#ifdef __linux__
+  if(data->set.mptcp)
+    transport = TRNSPRT_MPTCP;
```

Comment:
For the "Attention points" of the PR (see my previous comment above), maybe good to add:

```
- The `transport` is changed to `TRNSPRT_MPTCP` in `Curl_cf_tcp_create()`, not to change the logic before ("happy eyeballs", etc.), and because this feature is an extension to TCP, and it should only modify TCP connections. We are happy to change that if there is a better way.
```